diff -up vim-9.1.0888/src/configure.ac.3~ vim-9.1.0888/src/configure.ac
--- vim-9.1.0888/src/configure.ac.3~	2024-11-26 14:38:27.983967963 +0100
+++ vim-9.1.0888/src/configure.ac	2024-11-26 14:40:35.715608654 +0100
@@ -1177,7 +1177,7 @@ if test "$enable_perlinterp" = "yes" -o
       perllibs=`cd $srcdir; $vi_cv_path_perl -MExtUtils::Embed -e 'ldopts' | \
 		sed -e '/Warning/d' -e '/Note (probably harmless)/d' \
 			-e 's/-specs=[[^ ]*]//g' \
-			-e 's/-bE:perl.exp//' -e 's/-lc //'`
+			-e 's/-bE:perl.exp//' -e 's/-lc //' -e 's,-L[[^ ]]*/CORE ,,'`
       dnl Don't add perl lib to $LIBS: if it's not in LD_LIBRARY_PATH
       dnl a test in configure may fail because of that.
       dnl Remove -specs=<file-path>, the hardened flags cause relocation errors
