diff -up vim-8.1.0983/runtime/vimrc_example.vim.p25~ vim-8.1.0983/runtime/vimrc_example.vim
--- vim-8.1.0983/runtime/vimrc_example.vim.p25~	2019-02-26 22:31:56.251477350 +0100
+++ vim-8.1.0983/runtime/vimrc_example.vim	2019-02-26 22:32:39.246106616 +0100
@@ -40,6 +40,10 @@ augroup vimrcEx
   autocmd FileType text setlocal textwidth=78
 augroup END
 
+" Make shift-insert work like in Xterm
+map <S-Insert> <MiddleMouse>
+map! <S-Insert> <MiddleMouse>
+
 " Add optional packages.
 "
 " The matchit plugin makes the % command work better, but it is not backwards
