diff --git a/runtime/defaults.vim b/runtime/defaults.vim
index f3c639b..25f2707 100644
--- a/runtime/defaults.vim
+++ b/runtime/defaults.vim
@@ -77,13 +77,13 @@ inoremap <C-U> <C-G>u<C-U>
 " can position the cursor, Visually select and scroll with the mouse.
 " Only xterm can grab the mouse events when using the shift key, for other
 " terminals use ":", select text and press Esc.
-if has('mouse')
-  if &term =~ 'xterm'
-    set mouse=a
-  else
-    set mouse=nvi
-  endif
-endif
+# if has('mouse')
+#   if &term =~ 'xterm'
+#     set mouse=a
+#   else
+#     set mouse=nvi
+#   endif
+# endif
 
 " Only do this part when Vim was compiled with the +eval feature.
 if 1
