diff -up vim-9.0.1757/src/if_perl.xs.omv~ vim-9.0.1757/src/if_perl.xs
--- vim-9.0.1757/src/if_perl.xs.omv~	2023-08-20 15:28:57.452651937 +0200
+++ vim-9.0.1757/src/if_perl.xs	2023-08-20 15:29:39.687966463 +0200
@@ -709,7 +709,7 @@ S_POPMARK(pTHX)
 #  define Perl_POPMARK S_POPMARK
 # endif
 
-# if (PERL_REVISION == 5) && (PERL_VERSION >= 32)
+# if (PERL_REVISION == 5) && (PERL_VERSION >= 32) && (PERL_VERSION < 38)
 PERL_STATIC_INLINE U8
 Perl_gimme_V(pTHX)
 {
@@ -730,7 +730,7 @@ Perl_gimme_V(pTHX)
 }
 # endif
 
-# if (PERL_REVISION == 5) && (PERL_VERSION >= 38)
+# if (PERL_REVISION == 5) && (PERL_VERSION < 38)
 #  define PERL_ARGS_ASSERT_SVPVXTRUE             \
         assert(sv)
 PERL_STATIC_INLINE bool
