--- vim-8.0.1354/runtime/filetype.vim.p24~	2017-11-28 18:54:45.429518251 +0100
+++ vim-8.0.1354/runtime/filetype.vim	2017-11-28 18:55:09.649591235 +0100
@@ -1124,6 +1124,18 @@ au BufNewFile,BufRead *.pdf			setf pdf
 " PCMK - HAE - crm configure edit 
 au BufNewFile,BufRead *.pcmk 			setf pcmk
 
+" outline emacs file (TODO lists, ...)
+au BufNewFile,BufRead *		call FTCheck_outline()
+
+fun! FTCheck_outline()
+  " recognize Prolog by specific text in the first non-empty line
+  " require a blank after the '%' because Perl uses "%list" and "%translate"
+  let l = getline(nextnonblank(1))
+  if l =~ '-\*-\soutline\s-\*-'
+    setf outline
+  endif
+endfun
+
 " Perl
 if has("fname_case")
   au BufNewFile,BufRead *.pl,*.PL		call dist#ft#FTpl()
--- vim-8.0.1354/runtime/syntax/outline.vim.p24~	2017-11-28 18:54:45.429518251 +0100
+++ vim-8.0.1354/runtime/syntax/outline.vim	2017-11-28 18:54:45.429518251 +0100
@@ -0,0 +1,23 @@
+" Filename:    outline.vim
+" Purpose:     Vim syntax file
+" Language:    outline: emacs TODO lists
+" Maintainer:  Thierry Vignaud tvignaud@mandrakesoft.com
+" URL:         <none>
+" Last Change: Thu Oct 31 2002 <tvignaud@mandrakesoft.com>
+
+" For version 6.x: Quit when a syntax file was already loaded
+if exists("b:current_syntax")
+  finish
+endif
+
+"syn sync minlines=1000
+
+syn region EmacsMode oneline matchgroup=Comment start='-\*-\soutline\s-\*-' end='$'
+syn region FirstLevel oneline matchgroup=Function start='^\*[^*]' end='$'
+syn region SecondLevel oneline matchgroup=Structure start='^\*\*[^*]' end='$'
+syn region ThirdLevel oneline matchgroup=Statement start='^\*\*\*[^*]' end='$'
+syn region OtherLevels oneline matchgroup=Type start='^\*\{4,\}[^*]' end='$'
+
+let b:current_syntax = "outline"
+
+" vim: ts=8
