--- vim-8.0.0964/runtime/filetype.vim.orig	2017-08-19 19:05:20.262485714 +0200
+++ vim-8.0.0964/runtime/filetype.vim	2017-08-19 19:06:33.525310170 +0200
@@ -2821,6 +2821,8 @@ au BufNewFile,BufRead *.txt
 	\|   setf text
 	\| endif
 
+" dhcpd.conf
+au BufNewFile,BufRead dhcpd.conf                setf dhcpd
 
 " Use the filetype detect plugins.  They may overrule any of the previously
 " detected filetypes.
