diff -up vim-8.1.1471/runtime/gvim.desktop.omv~ vim-8.1.1471/runtime/gvim.desktop
--- vim-8.1.1471/runtime/gvim.desktop.omv~	2019-06-06 01:18:24.163141162 +0200
+++ vim-8.1.1471/runtime/gvim.desktop	2019-06-06 01:18:40.678141330 +0200
@@ -18,7 +18,6 @@ Comment=Edit text files
 # The translations should come from the po file. Leave them here for now, they will
 # be overwritten by the po file when generating the desktop.file!
 GenericName[da]=Teksteditor
-GenericName[eo]=Tekstoredaktilo
 GenericName[fr]=Éditeur de texte
 GenericName[pl]=Edytor tekstu
 GenericName[is]=Ritvinnsluforrit
@@ -37,7 +36,6 @@ Comment[da]=Rediger tekstfiler
 Comment[el]=Επεξεργασία αρχείων κειμένου
 Comment[en_CA]=Edit text files
 Comment[en_GB]=Edit text files
-Comment[eo]=Redakti tekstajn dosierojn
 Comment[es]=Edita archivos de texto
 Comment[et]=Redigeeri tekstifaile
 Comment[eu]=Editatu testu-fitxategiak
diff -up vim-8.1.1471/src/po/gvim.desktop.in.omv~ vim-8.1.1471/src/po/gvim.desktop.in
--- vim-8.1.1471/src/po/gvim.desktop.in.omv~	2019-06-06 01:27:12.525146532 +0200
+++ vim-8.1.1471/src/po/gvim.desktop.in	2019-06-06 01:27:42.933146841 +0200
@@ -10,7 +10,6 @@ Comment=Edit text files
 # The translations should come from the po file. Leave them here for now, they will
 # be overwritten by the po file when generating the desktop.file!
 GenericName[da]=Teksteditor
-GenericName[eo]=Tekstoredaktilo
 GenericName[fr]=Éditeur de texte
 GenericName[pl]=Edytor tekstu
 GenericName[is]=Ritvinnsluforrit
@@ -29,7 +28,6 @@ Comment[da]=Rediger tekstfiler
 Comment[el]=Επεξεργασία αρχείων κειμένου
 Comment[en_CA]=Edit text files
 Comment[en_GB]=Edit text files
-Comment[eo]=Redakti tekstajn dosierojn
 Comment[es]=Edita archivos de texto
 Comment[et]=Redigeeri tekstifaile
 Comment[eu]=Editatu testu-fitxategiak
