diff -up vim-8.1.2292/runtime/defaults.vim.omv~ vim-8.1.2292/runtime/defaults.vim
--- vim-8.1.2292/runtime/defaults.vim.omv~	2019-11-12 15:58:18.572807253 +0100
+++ vim-8.1.2292/runtime/defaults.vim	2019-11-12 16:00:09.318073189 +0100
@@ -77,7 +77,12 @@ inoremap <C-U> <C-G>u<C-U>
 " can position the cursor, Visually select and scroll with the mouse.
 " Only xterm can grab the mouse events when using the shift key, for other
 " terminals use ":", select text and press Esc.
-if has('mouse')
+"
+" However, enabling mouse support breaks cut&paste -- arguably the
+" cut&paste functionality provided by konsole and friends is more
+" useful than visually selecting. Let's limit mouse support to the
+" GUI version by default.
+if has('mouse') && has("gui_running")
   if &term =~ 'xterm'
     set mouse=a
   else
