diff -up vim-9.0.2073/src/xxd/xxd.c.2~ vim-9.0.2073/src/xxd/xxd.c
--- vim-9.0.2073/src/xxd/xxd.c.2~	2023-10-26 23:15:44.000000000 +0200
+++ vim-9.0.2073/src/xxd/xxd.c	2023-10-27 10:20:18.279143345 +0200
@@ -102,6 +102,7 @@
 #if __MWERKS__ && !defined(BEBOX)
 # include <unix.h>	/* for fdopen() on MAC */
 #endif
+#include <locale.h>
 
 
 /*  This corrects the problem of missing prototypes for certain functions
@@ -651,6 +652,12 @@ main(int argc, char *argv[])
   if (no_color == NULL || no_color[0] == '\0')
     color = enable_color();
 
+  char *lang=getenv("LANG");
+  if(!lang) lang=getenv("LC_ALL");
+  if(!lang) lang=getenv("LC_CTYPE");
+  if(lang)
+    setlocale(LC_ALL, lang);
+
 #ifdef AMIGA
   /* This program doesn't work when started from the Workbench */
   if (argc == 0)
@@ -1099,7 +1106,7 @@ main(int argc, char *argv[])
 #else
               (e > 31 && e < 127)
 #endif
-              ? e : '.';
+              ? e : isalnum(e) ? e : '.';
           n++;
           if (++p == cols)
             {
