diff -up vim-9.0.1833/src/xxd/xxd.c.2~ vim-9.0.1833/src/xxd/xxd.c
--- vim-9.0.1833/src/xxd/xxd.c.2~	2023-08-31 23:52:30.000000000 +0200
+++ vim-9.0.1833/src/xxd/xxd.c	2023-09-17 06:40:32.823417179 +0200
@@ -99,6 +99,7 @@
 #if __MWERKS__ && !defined(BEBOX)
 # include <unix.h>	/* for fdopen() on MAC */
 #endif
+#include <locale.h>
 
 
 /*  This corrects the problem of missing prototypes for certain functions
@@ -565,6 +566,12 @@ main(int argc, char *argv[])
   int addrlen = 9;
   int color = 0;
 
+  char *lang=getenv("LANG");
+  if(!lang) lang=getenv("LC_ALL");
+  if(!lang) lang=getenv("LC_CTYPE");
+  if(lang)
+    setlocale(LC_ALL, lang);
+
 #ifdef UNIX
   color = isatty(STDOUT_FILENO);
 #endif
@@ -1000,7 +1007,7 @@ main(int argc, char *argv[])
 #else
               (e > 31 && e < 127)
 #endif
-              ? e : '.';
+              ? e : isalnum(e) ? e : '.';
           n++;
           if (++p == cols)
             {
