diff -up vim-9.1.1917/src/xxd/xxd.c.2~ vim-9.1.1917/src/xxd/xxd.c
--- vim-9.1.1917/src/xxd/xxd.c.2~	2025-11-15 18:41:28.000000000 +0100
+++ vim-9.1.1917/src/xxd/xxd.c	2025-11-16 18:08:17.429729546 +0100
@@ -115,6 +115,7 @@
 #if __MWERKS__ && !defined(BEBOX)
 # include <unix.h>	/* for fdopen() on MAC */
 #endif
+#include <locale.h>
 
 
 /*  This corrects the problem of missing prototypes for certain functions
@@ -751,6 +752,12 @@ main(int argc, char *argv[])
   if (no_color == NULL || no_color[0] == '\0')
     color = enable_color();
 
+  char *lang=getenv("LANG");
+  if(!lang) lang=getenv("LC_ALL");
+  if(!lang) lang=getenv("LC_CTYPE");
+  if(lang)
+    setlocale(LC_ALL, lang);
+
 #ifdef AMIGA
   /* This program doesn't work when started from the Workbench */
   if (argc == 0)
@@ -1190,7 +1197,7 @@ main(int argc, char *argv[])
 #else
 	  (e > 31 && e < 127)
 #endif
-	  ? e : '.';
+	  ? e : isalnum(e) ? e : '.';
       n++;
       if (++p == cols)
 	{
