diff -up vim-9.0.0049/src/xxd/xxd.c.2~ vim-9.0.0049/src/xxd/xxd.c
--- vim-9.0.0049/src/xxd/xxd.c.2~	2022-07-09 21:27:07.000000000 +0200
+++ vim-9.0.0049/src/xxd/xxd.c	2022-07-10 05:16:38.676803818 +0200
@@ -98,6 +98,7 @@
 #if __MWERKS__ && !defined(BEBOX)
 # include <unix.h>	/* for fdopen() on MAC */
 #endif
+#include <locale.h>
 
 
 /*  This corrects the problem of missing prototypes for certain functions
@@ -502,6 +503,12 @@ main(int argc, char *argv[])
   char *varname = NULL;
   int addrlen = 9;
 
+  char *lang=getenv("LANG");
+  if(!lang) lang=getenv("LC_ALL");
+  if(!lang) lang=getenv("LC_CTYPE");
+  if(lang)
+    setlocale(LC_ALL, lang);
+
 #ifdef AMIGA
   /* This program doesn't work when started from the Workbench */
   if (argc == 0)
@@ -868,7 +875,7 @@ main(int argc, char *argv[])
 #else
 	  (e > 31 && e < 127)
 #endif
-	  ? e : '.';
+	  ? e : isalnum(e) ? e : '.';
       n++;
       if (++p == cols)
 	{
