diff -up vim-9.1.0888/src/xxd/xxd.c.2~ vim-9.1.0888/src/xxd/xxd.c
--- vim-9.1.0888/src/xxd/xxd.c.2~	2024-11-25 19:39:04.000000000 +0100
+++ vim-9.1.0888/src/xxd/xxd.c	2024-11-26 14:38:10.223737147 +0100
@@ -110,6 +110,7 @@
 #if __MWERKS__ && !defined(BEBOX)
 # include <unix.h>	/* for fdopen() on MAC */
 #endif
+#include <locale.h>
 
 
 /*  This corrects the problem of missing prototypes for certain functions
@@ -672,6 +673,12 @@ main(int argc, char *argv[])
   if (no_color == NULL || no_color[0] == '\0')
     color = enable_color();
 
+  char *lang=getenv("LANG");
+  if(!lang) lang=getenv("LC_ALL");
+  if(!lang) lang=getenv("LC_CTYPE");
+  if(lang)
+    setlocale(LC_ALL, lang);
+
 #ifdef AMIGA
   /* This program doesn't work when started from the Workbench */
   if (argc == 0)
@@ -1141,7 +1148,7 @@ main(int argc, char *argv[])
 #else
               (e > 31 && e < 127)
 #endif
-              ? e : '.';
+              ? e : isalnum(e) ? e : '.';
         }
         n++;
         if (++p == cols)
