diff -up vim-8.1.1471/src/xxd/xxd.c.xxdloc~ vim-8.1.1471/src/xxd/xxd.c
--- vim-8.1.1471/src/xxd/xxd.c.xxdloc~	2019-06-05 22:57:04.000000000 +0200
+++ vim-8.1.1471/src/xxd/xxd.c	2019-06-06 01:10:11.403136154 +0200
@@ -95,6 +95,7 @@
 #if __MWERKS__ && !defined(BEBOX)
 # include <unix.h>	/* for fdopen() on MAC */
 #endif
+#include <locale.h>
 
 
 /*  This corrects the problem of missing prototypes for certain functions
@@ -468,6 +469,12 @@ main(int argc, char *argv[])
   char *pp;
   int addrlen = 9;
 
+  char *lang=getenv("LANG");
+  if(!lang) lang=getenv("LC_ALL");
+  if(!lang) lang=getenv("LC_CTYPE");
+  if(lang)
+         setlocale(LC_ALL, lang);
+
 #ifdef AMIGA
   /* This program doesn't work when started from the Workbench */
   if (argc == 0)
@@ -854,7 +861,7 @@ main(int argc, char *argv[])
 #else
 	  (e > 31 && e < 127)
 #endif
-	  ? e : '.';
+	  ? e : isalnum(e) ? e : '.';
       n++;
       if (++p == cols)
 	{
